contracts/src/CrossChainApplications/examples/NativeTokenBridge/NativeTokenDestination.sol

```diff
@@ -37,17 +36,29 @@ contract NativeTokenDestination is TeleporterOwnerUpgradeable, INativeTokenDesti
      *
      * @dev Defined as BLACKHOLE_ADDRESS at
      * https://github.com/ava-labs/subnet-evm/blob/e23ab058d039ff9c8469c89b139d21d52c4bd283/constants/constants.go
+     * It is a system-reserved address by default in subnet-evm, and transfers cannot be sent here manually.
      */
     address public constant BURNED_TX_FEES_ADDRESS = 0x0100000000000000000000000000000000000000;
 
     /**
-     * @notice Designated Blackhole Address for this contract.
+     * @notice Address used by this contract to blackhole funds, effectively burning them.
      *
-     * @dev Tokens are sent here to be "burned" before sending an unlock message to the source chain. Different
-     * from the burned tx fee address so they can be tracked separately.
+     * @dev Native tokens are burned by this contract by sending them to this address when transferring tokens back to
+     * the source chain. Different from BURNED_TX_FEES_ADDRESS so that the total amount burned in transaction fees and
+     * the total amount burned to be sent back to the source chain can be tracked separately.
+     * This address was chosen arbitrarily.
      */
     address public constant BURN_FOR_TRANSFER_ADDRESS = 0x0100000000000000000000000000000000010203;
 
+    /**
+     * @notice Address used by the source chain to blackhole funds, effectively burning them.
+     *
+     * @dev When reporting burned transaction fee amounts, this address is used as the recipient
+     * address for the funds to be sent to be burned on the source chain.
+     * This address was chosen arbitrarily.
+     */
+    address public constant SOURCE_CHAIN_BURN_ADDRESS = 0x0100000000000000000000000000000000010203;
```

Comment:
do we need this duplicate addrses variable? Can we just use `BURN_FOR_TRANSFER_ADDRESS` in reporting burned fees

Reply:
https://github.com/ava-labs/teleporter/pull/347#discussion_r1523699992